src/drivers/imx/sai.c

```diff
@@ -264,6 +308,10 @@ static inline int sai_set_config(struct dai *dai,
 
 	val_cr2 |= REG_SAI_CR2_SYNC;
 	mask_cr2 |= REG_SAI_CR2_SYNC_MASK;
+
+	/* TODO: for the time being use half FIFO size as watermark */
+	dai_update_bits(dai, REG_SAI_XCR1(REG_RX_DIR),
+			REG_SAI_CR1_RFW_MASK, SAI_FIFO_WORD_SIZE / 2);
 	dai_update_bits(dai, REG_SAI_XCR2(REG_RX_DIR), mask_cr2, val_cr2);
 	dai_update_bits(dai, REG_SAI_XCR4(REG_RX_DIR), mask_cr4, val_cr4);
```

Comment:
Ok, so you move FIFO_WORD_SIZE setting from sai_start to sai_set_config? I can understand this change but you need to add it in a separate patch and explain why is it needed. It doesn't look to be related to the current change.

Reply:
Yes, I've missed this change. I'll add a new commit.
There's no need to set the FIFO size each time, on start.

Reply:
@iuliana-prodan you can add a new commit. That's an obvious change and if you want to be merged faster we can send it in a separate PR. 
For the rest of the changes lets give Intel people some time to look at this. I will also want to take my time to make some suggestions by the end of the week.

Reply:
@dbaluta I've added a new commit just for the SAI watermark.